src/stores/connectors/ethereum/list.go

```diff
@@ -8,40 +8,40 @@ import (
 	"github.com/ethereum/go-ethereum/common"
 )
 
-func (c Connector) List(ctx context.Context) ([]common.Address, error) {
+func (c Connector) List(ctx context.Context, limit, offset int) ([]common.Address, error) {
 	err := c.authorizator.CheckPermission(&types.Operation{Action: types.ActionRead, Resource: types.ResourceEthAccount})
 	if err != nil {
 		return nil, err
 	}
 
-	accs, err := c.db.GetAll(ctx)
+	strAddr, err := c.db.ListAddresses(ctx, false, limit, offset)
 	if err != nil {
 		return nil, err
 	}
 
 	addrs := []common.Address{}
-	for _, acc := range accs {
-		addrs = append(addrs, acc.Address)
+	for _, addr := range strAddr {
+		addrs = append(addrs, common.HexToAddress(addr))
 	}
 
 	c.logger.Debug("ethereum accounts listed successfully")
 	return addrs, nil
 }
 
-func (c Connector) ListDeleted(ctx context.Context) ([]common.Address, error) {
+func (c Connector) ListDeleted(ctx context.Context, limit, offset int) ([]common.Address, error) {
 	err := c.authorizator.CheckPermission(&types.Operation{Action: types.ActionRead, Resource: types.ResourceEthAccount})
 	if err != nil {
 		return nil, err
 	}
 
-	accs, err := c.db.GetAllDeleted(ctx)
+	strAddr, err := c.db.ListAddresses(ctx, true, limit, offset)
```

Comment:
Can we rename `ListAddresses` to `SearchAddresses` ? like we have in Orchestrate. Seems better to me.

Reply:
But we are not doing any search, we are actually just listing. In Orchestrate we named "Search" because it was an actual search method which return a list of items, in this case we are listing a ids/addresses

Reply:
But I believe we will use the same function to Search if we have to in the future. Searching will return a list of IDs in the QKM. 

We will pass filters to this function to achieve this. Limit and offset are examples of filters already and you are searching for specific IDs.

Reply:
I see your point, but I barely see this function will evolve that way. Lets modify it though and let see :)